openapi/sled-agent.json

```diff
@@ -3232,6 +3232,13 @@
               "$ref": "#/components/schemas/DiskRequest"
             }
           },
+          "boot_order": {
```

Comment:
whatever we do with the user-facing API, i'm going to keep the `sled-agent` API closer to Propolis and what we with [`BootOrder`](https://github.com/oxidecomputer/propolis/pull/756#discussion_r1761872738).